lib/httpRequestBuilder.js

```diff
@@ -13,12 +13,54 @@ function requestBuilder (defaults) {
     headers: {},
     body: Buffer.alloc(0),
     hostname: 'localhost',
+    chunkSize: 0,
     setupRequest: reqData => reqData,
     port: 80
   }
 
   defaults = Object.assign(builderDefaults, defaults)
 
+  const chunkBody = (body, size) => {
+    if (size === 0) {
+      return body
+    }
+    const chunks = []
+    let found = body.search(/\[<id>\]/g)
+    found = found === -1 ? false : found
+    let start = 0
+    while (start < body.length) {
+      if (found && (start + size) > found) {
+        chunks.push(body.substr(start, found - start))
+        chunks.push('[<id>]')
+        start = found + '[<id>]'.length
+        found = false
+        continue
+      }
+      chunks.push(body.substr(start, size))
+      start = start + size
+    }
```

Comment:
This is going to be _expensive_. Can we run it once and cache it?

Reply:
Thanks for the feedback, since you seem to be happy with the direction, I will now keep on writing it and improving.